### PR TITLE
Add rails integration testing build configuration to circleci

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
1.  Add new `lms_integration_build` to circleci configuration to run Rails [system specs](https://medium.com/table-xi/a-quick-guide-to-rails-system-tests-in-rspec-b6e9e8a8b5f6) utilizing a headless browser.
2. Remove `database_cleaner` gem as it is no longer [needed](https://stackoverflow.com/a/49248170)
3. Remove `poltergeist` configuration in favor of `capybara-selenium`.
4. Add `capybara-screenshot` gem and use output to store artifacts from failed browser tests.

## WHY
System specs are high-level integration specs that check for coordination.

During an upgrade from Rails 5.0 -> 5.1 [three](https://www.notion.so/quill/Connect-SentryError-Api-V1-ActiveActivitySessionsController-update-ActionController-Unfilter-d4e5b636581e43d9a1c5d11fab66db42) [critical](https://www.notion.so/quill/Proofreader-Grammar-SentryError-Api-V1-ActiveActivitySessionsController-update-ActionContro-8f550de7f40f40658f8cd55a73348a0e) errors were not caught with the existing test suite since the scope is limited to controller testing.  Adding a framework for system testing can ensure happy path flows from a high level are working properly after upgrades.

## HOW

Initially the scope of this end-to-end test was a user [login](https://github.com/empirical-org/Empirical-Core/blob/423055a21ef95ece37f9678ddec533f7f06056ad/services/QuillLMS/spec/system/google_auth_spec.rb#L8) through google auth.  Unfortunately, this route was [blocked](https://stackoverflow.com/a/20594894) by Google adding  [2FA](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/10088/workflows/c15c0c26-44bc-4ec3-b987-9938ebe63ad8/jobs/173670/artifacts), and [captchas](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/10088/workflows/d22c9d33-41c3-44e7-8de3-78e214d34de6/jobs/173605/artifacts) in order to block automated logins.  They also use [A/B testing](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/10074/workflows/0247ae61-920b-46b8-b08e-5b599f0e5cd4/jobs/173507/artifacts) on the login prompts which complicates the locating of page elements. The scope was scaled back to a simple user login within the Quill user base.

1.  Starting with the existing `lms_rails_build` configuration, create a new build and add in npm dependencies since webpack needs to build assets for systems specs.
2.  [Configure](https://github.com/empirical-org/Empirical-Core/pull/7977/files#diff-38aa2535583194633c91feb14beb3e57e293132235aef93ab732c31b2a1c09feR34) capybara driver to use chrome both locally and on remote (i.e. circleci) environments.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Add-rails-integration-testing-build-configuration-to-circleci-d3d6606d478d461d928f11336e36d593

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Design Review: If applicable, have you compared the coded design to the mockups? | N/A

### CODE REVIEW

**Comment by brendanshean** [comment]
Example of context usage:

```
workflows:
  version: 2
  build-test:
    jobs:
      - lms_rails_build:
          context:
            - google_client_config
            - google_test_user
```

**Review by happythenewsad — APPROVED** [review]
Approving with comment:

~This adds almost 5 minutes to our CI builds, which is significant: https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core?branch=bs-integration-testing~

My mistake - I thought that was 5 mins sequential time, not parallel time

**Comment by brendanshean** [comment]
> Approving with comment:
> 
> This adds almost 5 minutes to our CI builds, which is significant: https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core?branch=bs-integration-testing

Originally, I had the integration tests run within the `lms_rails_build`, but runtime ballooned to upwards of 6 or 7 minutes.  Moving the integration specs to a separate build gives us the benefit of concurrent builds so it's not adding too much more time:

Screen Shot 2021-07-28 at 4 54 06 PM

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/spec/support/system.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+RSpec.configure do |config|
+  Capybara.register_driver :local_selenium_chrome_headless do |app|
+    options = Selenium::WebDriver::Chrome::Options.new(
+      args: [
+        'headless',
```
Comment (dandrabik) [inline services/QuillLMS/spec/support/system.rb:5]: 🧟

**Review by dandrabik — APPROVED** [review]
This looks good to me. Really excited about having these tests in place.

:100: :100:

**Pull request merged**